File: algorithms/bubble-sort/python/bubblesort.py
#!/usr/bin/env python3

def bubble_sort(arr):
  n = len(arr)
  for i in range(n):
    for j in range(1, n - i):
      if arr[j - 1] > arr[j]:
        arr[j - 1], arr[j] = arr[j], arr[j - 1]

if __name__ == "__main__":
  arr = [6, 3, 5, 10, -1, 7, -4]
  bubble_sort(arr)

File: algorithms/bubble-sort/c/bubblesort.c
#include <stdio.h>

void swap(int *a, int *b)
{
  int t = *a;
  *a = *b;
  *b = t;
}

void bubble_sort(int *a, int n)
{
  for (int i = 0; i < n; i++) {
    for (int j = 1; j < n - i; j++) {
      if (a[j - 1] > a[j]) {
        swap(&a[j - 1], &a[j]);
      }
    }
  }
}

int main(void)
{
  int a[] = {5, -1, 3, 8, -4, 0};
  bubble_sort(arr, 6);
  return 0;
}

